src/components/ui/dropdown.svelte

```diff
@@ -1,31 +1,55 @@
 <script>
+  import { createEventDispatcher } from 'svelte';
   import Button from './button.svelte';
 
+  export let classname = 'dropdown-shell';
+  export let btnclass = 'handle';
+  export let dropdownclass = 'dropdown';
+  export let wrapperclass = 'relative-wrapper';
+  export let isRight = false;
+
   export let label = '';
-  export let right = false;
   export let chevron = true;
-  export let btnclass = '';
-  export let dropdownclass = '';
-  export let wrapperclass = '';
   export let nowrap = false;
-  export let open = false;
-  export const customLabel = false;
-  export const toggle = () => (open = !open);
+  export let value = false;
+
+  $: isOpen = value;
+  let justOpened = false;
+  const dispatch = createEventDispatcher();
+  const toggle = () => {
+    isOpen = !isOpen;
+    dispatch('change', isOpen);
+    if (isOpen) justOpened = true;
+  };
+
+  let dropNode = null;
+  const clickOutside = (event) => {
+    let isClickInside = dropNode.contains(event.target);
+    if (!isClickInside) {
+      if (justOpened) justOpened = false;
+      else if (isOpen) {
+        isOpen = false;
+        dispatch('change', isOpen);
+      }
+    }
+  };
+
 </script>
 
-<div class:open class="dropdown-shell">
-  <Button on:click={toggle} classname="handle {btnclass}">
+<svelte:window on:click={clickOutside} />
+<div class:open={isOpen} class={classname}>
+  <Button on:click={toggle} classname="{btnclass} btn">
     <slot name="label">{label}</slot>
     {#if chevron}
       <svg src="images/icons/chevron-down.svg" class="icon ml chevron" />
     {/if}
   </Button>
-  <div class:right-edge={right} class="dropdown {dropdownclass}">
+  <div class:right-edge={isRight} class={dropdownclass} bind:this={dropNode}>
     {#if nowrap}
       <slot />
     {:else}
-      <div class="relative-wrapper {wrapperclass}">
-        <Button on:click={toggle} normal round classname="close">
+      <div class={wrapperclass}>
```

Comment:
Wrapping should add padding to content because it gets clipped at the corners. The close button should not affect the content.
![image](https://user-images.githubusercontent.com/11016151/70454172-0eef6d00-1abb-11ea-8f20-12515eaae235.png)

Reply:
Css-oriented problem is for the markup repo